airflow/providers/openlineage/CHANGELOG.rst

```diff
@@ -26,10 +26,10 @@
 Changelog
 ---------
 
-main
+1.8.0
 .....
 
-In Airflow 2.10.0, we fix the way try_number works, so that it no longer returns different values depending on task instance state.  Importantly, after the task is done, it no longer shows current_try + 1. Thus in 1.7.2 we patch this provider to fix try_number references so they no longer adjust for the old, bad behavior.
+In Airflow 2.10.0, the way try_number works changed, so that it no longer returns different values depending on task instance state. Importantly, after the task is done, it no longer shows current_try + 1. Thus in 1.8.0 version of this provider we fix try_number references so they no longer adjust for the old, bad behavior. This change makes all previous versions of OpenLineage provider (<1.8.0) incompatible with Airflow >= 2.10.0.
```

Comment:
If I am user that wants to upgrade.
What is the action required from me?
This text explain what was done to the code base. It does not explain what I need to do on upgrade

Reply:
Simple "You need to use openlineage provider version 1.8.0 with Airflow 2.10.0+ if you want to integrate Open Lineage with Airflow". Yes. I agree that our changelog entries should be more empathetic towards the user. Probably good idea - in such case is to explain first what the user should pay attention to (version of the open lineage provider) and when (when they are using openlineage provider) and then why.

BTW. This however should be a neswfragment in arflow newsfragments/ not in provider. I presume (as discussed before) 1.8 version of the provider will add (in the next PR) a way to handle try_number when installed on previous version of Airlfow.

If that's the case, then provider would not even have to technically be breaking change (assuming full backwards compatibility with previous Airflow versions) but I would still leave 1.8.0 to demarkate this "big" change better.

Reply:
> It's already [there](https://github.com/apache/airflow/blob/6bcec908cd27d691e97156a29c7243db9089e0f8/airflow/providers/openlineage/plugins/listener.py#L49), there is a function that determines whether we should subtract one from try_number based on Airflow version

Right :)